src/poetry/installation/executor.py

```diff
@@ -582,14 +582,25 @@ def _prepare_git_archive(self, operation: Install | Update) -> Path:
         from poetry.vcs.git import Git
 
         package = operation.package
+        reference = package.source_resolved_reference or package.source_reference
+        assert package.source_url is not None
+        assert reference is not None
+
+        cache_dir = self._chef.get_cache_directory_for_git(
+            package.source_url, reference
+        )
+        cache_dir.mkdir(parents=True, exist_ok=True)
+        cached_archive = next(cache_dir.glob("*.whl"), None)
+        if cached_archive is not None and not operation.package.develop:
+            return cached_archive
```

Comment:
We should extract this to method in `ArtifactCache` similar to `get_cached_archive_for_link`.